plugins/coreboot/fu-plugin-coreboot.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2019 9elements Agency GmbH <patrick.rudolph@9elements.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+#include <stdio.h>
+#include <glib.h>
+#include <glib/gstdio.h>
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-device-metadata.h"
+#include "fu-device-private.h"
+#include "fu-plugin-coreboot.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_SUPPORTS_PROTOCOL, "org.coreboot.fmap");
+
+	/* make sure that flashrom plugin is ready to receive devices */
+	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_RUN_AFTER, "org.flashrom");
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+void
+fu_plugin_destroy (FuPlugin *plugin)
+{
+}
+
+gboolean
+fu_plugin_coldplug (FuPlugin *plugin, GError **error)
+{
+	g_autofree const gchar *triplet = NULL;
+	g_autofree gchar *name = NULL;
+	const gchar *major;
+	const gchar *minor;
+	const gchar *vendor;
+	const gchar *version;
+	g_autoptr(FuDevice) dev;
+	GBytes *bios_table;
+	gboolean updatable = TRUE;
+	gint firmware_size = 0;
+	struct fmap *fmap;
+
+	vendor = fu_plugin_get_dmi_value (plugin, FU_HWIDS_KEY_BIOS_VENDOR);
+	if (!vendor)
+		return FALSE;
+
+	version = fu_plugin_get_dmi_value (plugin, FU_HWIDS_KEY_BIOS_VERSION);
+	if (version) {
+		triplet = fu_plugin_coreboot_version_string_to_triplet (version, error);
+		/* Ignore developer builds */
+		if (g_strrstr (version, "dirty"))
```

Comment:
Why ignore developer builds?  I think they're the most likely to want to use this plugin!  I would think it makes sense to still allow flashing in this instance but require `--force` instead.

Reply:
Needs to be investigated if it can be enabled by default. Developers still can flash using flashrom.